components/elm/src/cpl/lnd_disagg_forc.F90

```diff
@@ -0,0 +1,797 @@
+    
+    ! Do the downscaling
+    hsurf_g = grc_pp%elevation(g) !ldomain%topo2(g)
+    hsurf_t = top_pp%elevation(t)
+
+    ! Here we assume that deltaLW = (dLW/dT)*(dT/dz)*deltaz
+    ! We get dLW/dT = 4*eps*sigma*T^3 = 4*LW/T from the Stefan-Boltzmann law,
+    ! evaluated at the mean temp.
+    ! We assume the same temperature lapse rate as above.
+
+    tair_g = x2l(index_x2l_Sa_tbot,i)
+    tair_t = top_as%tbot(t)
+    tsfc_g = lnd2atm_vars%t_rad_grc(g)
+    lwrad_g = x2l(index_x2l_Faxa_lwdn,i)
+    if (method == 0 .or. (method == 1 .and. nstep == 0)) then
+       lwrad_t = lwrad_g - &
+          4.0_r8 * lwrad_g/(0.5_r8*(tair_t+tair_g)) * &
+          lapse_glcmec * (hsurf_t - hsurf_g)
+    else
+       
+       lwrad_t = lwrad_g + lnd2atm_vars%eflx_lwrad_out_grc(g) * &
+          4._r8 * (tair_t - tair_g) / tsfc_g
+       
+    end if
+       top_af%lwrad(t) = lwrad_t
+
+  end subroutine downscale_longwave_to_topounit
+  
+  !-------------------------------------------------------
+  ! Downscale precipitation from grid to topounit using FNM method
+  subroutine downscale_precip_to_topounit_FNM(mxElv,uovern_t,grdElv,topoElv,rain_g,snow_g,deltaR,deltaS,hrise) 
+   
+    ! !ARGUMENTS:
+    real(r8) :: rain_g, snow_g
+    real(r8) :: mxElv                    ! Maximum elevation value per grid
+    real(r8) :: uovern_t                 ! Froude Number
+    real(r8) :: grdElv                   ! Grid elevation
+    real(r8) :: topoElv                  ! Topounit elevation
+    real(r8), intent(inout) :: deltaR
+    real(r8), intent(inout) :: deltaS
+    real(r8), intent(inout) :: hrise    
+    
+    ! !LOCAL VARIABLES:
+    real(r8) :: r
+    real(r8) :: topoEl
+    real(r8) :: mxEl 
+    real(r8) :: grdEl
+    character(len=*), parameter :: subname = 'downscale_precip_to_topounit_FNM'
+    !--------------------------------------------------------------------------------------
+    
+    if (mxElv < 0) then
+       mxEl = abs(mxElv) ! avoid situations where -ve r can force lower areas more rain than higher elevation areas.
+       topoEl = abs(topoElv)
+       grdEl = abs(grdElv)
+    else 
+       mxEl = mxElv ! avoid situations where -ve r can force lower areas more rain than higher elevation areas.
+       topoEl = topoElv
+       grdEl = grdElv
+    end if    
+    
+    r = 0.  
+    r = topoEl - grdEl
+    if (uovern_t < 0) then
+       hrise = r
+    else
+       hrise = min(uovern_t,r)
+    end if    
+    
+    if (rain_g < 0) then
+       deltaR = 0.
+    else          
+       deltaR = (rain_g*(hrise/mxEl))
+    end if
+
+    if (snow_g < 0) then
+       deltaS = 0. 
+    else
+       deltaS = (snow_g*(hrise/mxEl))
+    end if
+  end subroutine downscale_precip_to_topounit_FNM  
+  
+  !-------------------------------------------------------
+  ! Downscale precipitation from grid to topounit using ERMM method
+  subroutine downscale_precip_to_topounit_ERMM(t,mxElv,grdElv,topoElv,rain_g, snow_g,elv_flag,elvrnge) 
```

Comment:
Add comment here to explain what ERMM method is

Reply:
@thorntonpe I have added comments describing ERMM in my latest commit. Thanks.